### ISSUE TITLE
SPDX - invalid LicenseRef

### ISSUE THREAD

**Issue by bobgob** [fossology/fossology#492]
1) Upload and scan the following:
http://mirror.sdunix.com/gnu/time/
http://mirror.sdunix.com/gnu/cpio/cpio-2.10.tar.bz2
http://git.spdx.org/?p=spdx-tools.git;a=snapshot;h=daa3f134942b11b7839d4cad6e151a65c9ae3be1;sf=tgz

2) Select all 3 in the License Browser and click on "SPDX Generation"

3) Verify
(You can get the SPDXTools Version 2.0.2 from https://spdx.org/tools/spdx/consolidated-spdx-tools-and-library ).  Note this means we need to change a few license names or request a larger character set from SPDX.

```
bobg$ java -jar spdx-tools-2.0.2-jar-with-dependencies.jar Verify SPDX2_time_1439848453.rdf 
 WARN [main] (ListedLicenses.java:342) - Unable to access the SPDX listed licenses at http://www.spdx.org/licenses.  Using local file copy of SPDX listed licenses
This SPDX Document is not valid due to:
	Invalid license id 'LicenseRef-Gov't-work'.  Must start with 'LicenseRef-' and made up of the characters from the set 'a'-'z', 'A'-'Z', '0'-'9', '+', '_', '.', and '-'.
	Invalid license id 'LicenseRef-JPEG/netpbm'.  Must start with 'LicenseRef-' and made up of the characters from the set 'a'-'z', 'A'-'Z', '0'-'9', '+', '_', '.', and '-'.
	Invalid license id 'LicenseRef-See-doc(OTHER)'.  Must start with 'LicenseRef-' and made up of the characters from the set 'a'-'z', 'A'-'Z', '0'-'9', '+', '_', '.', and '-'.
	Invalid license id 'LicenseRef-Non-profit!'.  Must start with 'LicenseRef-' and made up of the characters from the set 'a'-'z', 'A'-'Z', '0'-'9', '+', '_', '.', and '-'.
```

**Comment by pguttmann** [comment]
According to Appendix IV of http://spdx.org/sites/spdx/files/SPDX-2.0.pdf I'd suggest dropping the ' in Gov't, change / to . in JPEG/netpbm, change (OTHER) to .OTHER and drop ! in Non-profit! as follows:
'LicenseRef-Govt-work'
'LicenseRef-JPEG.netpbm'
'LicenseRef-See-doc.OTHER'
'LicenseRef-Non-profit'

**Comment by bobgob** [comment]
Email was sent to fossology-devel identifying ALL the licenses that need changing.  Did you see it?

**Comment by pguttmann** [comment]
Oops - yes I did and clicked on the first link which brought me here, *before* I read the body ;)
Based upon the SPDX 2.0 spec I linked to above, I no longer see '+' or '_' as allowed characters ;)
I guess that would change some of your options ;)
I agree with your suggestions with that caveat regarding underscore.

**Comment by bobgob** [comment]
Indeed.  I'll write spdx for clarification since the spec differs from their validation tool

**Comment by mcjaeger** [comment]
yes, it is in the fisst section on page 82 of http://spdx.org/sites/spdx/files/SPDX-2.0.pdf
Is there an answer maybe from the SPDX?

**Comment by pguttmann** [comment]
It's not listed on page 81 though ;) See the idstring "definition".

According to pages 81 and 82 it appears to be available *only* as an optional suffix character for SPDX "canned" license-id strings, not for user defined idstrings.

**Comment by steffen-weber** [comment]
The http://spdx.org/rdf/terms do not restrict '+' as suffix: http://spdx.org/rdf/terms/dataproperties/licenseId___-500276407.html
Assuming that '+' is only allowed as suffix, invalid shortnames can be found by ``SELECT rf_shortname FROM license_ref WHERE rf_shortname !~* E'^[[:alnum:].-]*[+]?$'``.
For "GPL-2.0+-with-classpath-exception" et al., clarification is required.

**Comment by mcjaeger** [comment]
just an update: I think there is a pull request, but a migration script is missing.

**Comment by mcjaeger** [comment]
Bob has merged pull #526 already, let's check of the migration works on different instances

**Comment by mcjaeger** [comment]
closing issue, becaue in fact I was sleeping writing the above, since we will do th rc-phase (release candidate)
